### DISCUSSION TITLE
Can't use next/link in a component library

### DISCUSSION THREAD

**Discussion by mattiaz9** [vercel/next.js#37307] [Help]
I'm building a component library for nextjs, and I have a Link component (which is a wrapper around the nextjs Link).

When I try to click on my Link component I get this error: `TypeError: null is not an object (evaluating 'router[replace ? 'replace' : 'push']')`.

Looks like the nextjs Link component can't handle routing when imported from an external library. 

Does anyone know how to solve this issue?

**Comment by icyJoseph** [comment]
It sounds like the component is transpiled/compiled in the external library, in absence of the Router Context Provider. 

Would it be possible to take a look at some code?

**Reply by mattiaz9** [reply]
Hi @icyJoseph ! Yes, it's transpiled using Rollup. 

Do you know if it's possible to make it work with transpiling? 
Personally I wouldn't mind to keep the components in the library un-transpiled, but if I do so I get this error instead:
```
ModuleParseError: Module parse failed: Unexpected token (8:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See [https://webpack.js.org/concepts#loaders](https://webpack.js.org/concepts#loaders|)
```
